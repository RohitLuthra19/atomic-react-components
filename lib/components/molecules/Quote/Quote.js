// @flow
import React from 'react';
import type { Node } from 'react';
import styled from 'styled-components';
import classnames from 'classnames';
import styles from './Quote.style';

type Props = {
  className: string,
  content: string,
  contentSrc: string,
  author?: string,
  company?: string,
};

const Quote = ({ content, contentSrc, author, company, className, ...otherProps }: Props): Node => {
  return (
    <div {...otherProps} className={classnames('quote-wrap', className)}>
      <div className="quote__mark">&rdquo;</div>
      <blockquote className="quote__content" cite={contentSrc}>
        <p>{content}</p>
        <footer className="quote__author">
          {typeof author !== 'undefined' ? <cite>{author}</cite> : ''}
          {typeof author !== 'undefined' && typeof company !== 'undefined' ? <span>, </span> : ''}
          {typeof company !== 'undefined' ? <cite>{company}</cite> : ''}
        </footer>
      </blockquote>
    </div>
  );
};

Quote.defaultProps = {
  author: undefined,
  company: undefined,
};

export default styled(Quote)`
  ${styles};
`;
export { Quote as QuoteVanilla };
