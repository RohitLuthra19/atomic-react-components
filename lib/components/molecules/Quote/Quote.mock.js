const defaultQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  author: 'Stephen Covey',
  company: 'My company',
  contentStyle: {
    color: 'red',
  },
  authorStyle: { color: 'blue' },
  quoteStyle: {},
};

const noAuthorQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  company: 'My company',
  contentStyle: {},
  authorStyle: {},
  quoteStyle: {},
};

const noCompanyQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  author: 'Stephen Covey',
  contentStyle: {},
  authorStyle: {},
  quoteStyle: {},
};

const noCompanyNoAuthorQuote = {
  className: 'quote',
  content: `Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.`,
  contentSrc: 'https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list',
  contentStyle: {},
  authorStyle: {},
  quoteStyle: {},
};


export { defaultQuote, noAuthorQuote, noCompanyQuote, noCompanyNoAuthorQuote };
