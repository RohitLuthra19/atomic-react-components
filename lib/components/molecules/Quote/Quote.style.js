import { css } from 'styled-components';

const styles = css`
  .quote {
    &__mark {
      font-size: 60px;
      float: left;
    }
    &__content {
      padding-top: 5px;
      ${props => (props.contentStyle ? props.contentStyle : '')};
    }

    &__author {
      float: right;
      ${props => (props.authorStyle ? props.authorStyle : '')};
    }
    ${props => (props.quoteStyle ? props.quoteStyle : '')};
  }
`;

export default styles;
