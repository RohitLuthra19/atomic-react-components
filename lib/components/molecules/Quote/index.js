// @flow

export { default } from './Quote';
export { QuoteVanilla } from './Quote';
