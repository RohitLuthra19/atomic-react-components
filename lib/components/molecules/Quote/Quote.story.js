// @flow
import React from 'react';
import { storiesOf } from '@storybook/react';

import Quote, { QuoteVanilla } from './Quote';
import { defaultQuote, noAuthorQuote, noCompanyQuote } from './Quote.mock'

// $FlowFixMe eslint-disable-next-line extra-rules/no-commented-out-code
storiesOf('Molecules', module).addWithChapters('Quote', {
  chapters: [
    {
      sections: [
        {
          sectionFn: () => (
            <QuoteVanilla {...defaultQuote}
         
            />
          ),
          options: {
            showSource: true,
            allowSourceToggling: true,
            showPropTables: true,
            allowPropTablesToggling: true,
          },
        },
      ],
    },
    {
      title: 'Quote with all fields',
      sections: [
        {
          title: 'With all fields',
          sectionFn: () => <Quote {...defaultQuote} />,
        },
        
      ],
    },
    {
      title: 'Quote without author',
      sections: [
        {
          title: 'No Author and company present',
          sectionFn: () => <Quote {...noAuthorQuote} />,
        },
        
      ],
    },
    {
      title: 'Quote without company',
      sections: [
        {
          title: 'No Company but author present',
          sectionFn: () => <Quote {...noCompanyQuote} />,
        },
        
      ],
    },
  ],
});
