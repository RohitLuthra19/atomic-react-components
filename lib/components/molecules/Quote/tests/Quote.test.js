import React from 'react';
import { shallow, mount } from 'enzyme';
import { QuoteVanilla } from '../index';
import { defaultQuote, noAuthorQuote, noCompanyQuote } from '../Quote.mock';

describe('<Quote /> Rendering', () => {
  test('should render correctly with all fields', () => {
    const quote = shallow(<QuoteVanilla {...defaultQuote} />);
    expect(quote).toMatchSnapshot();
  });

  test('should render correctly with no author', () => {
    const quote = shallow(<QuoteVanilla {...noAuthorQuote} />);
    expect(quote).toMatchSnapshot();
  });

  test('should render correctly with no company', () => {
    const quote = shallow(<QuoteVanilla {...noCompanyQuote} />);
    expect(quote).toMatchSnapshot();
  });

  test('should render correctly with all fields', () => {
    const quote = mount(<QuoteVanilla {...defaultQuote} />);

    expect(quote.find('.quote__author').text()).toEqual('Stephen Covey, My company');
  });

  test('should render correctly with no author', () => {
    const quote = mount(<QuoteVanilla {...noAuthorQuote} />);
    console.log('ghost');
    console.log(quote.find('.quote__author'));
    expect(quote.find('.quote__author').text()).toEqual('My company');
  });

  test('should render correctly with no company', () => {
    const quote = mount(<QuoteVanilla {...noCompanyQuote} />);
    expect(quote.find('.quote__author').text()).toEqual('Stephen Covey');
  });
});
