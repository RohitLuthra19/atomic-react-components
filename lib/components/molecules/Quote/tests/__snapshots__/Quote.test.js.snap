// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Quote /> Rendering should render correctly with all fields 1`] = `
<div
  authorStyle={
    Object {
      "color": "blue",
    }
  }
  className="quote-wrap quote"
  contentStyle={
    Object {
      "color": "red",
    }
  }
  quoteStyle={Object {}}
>
  <div
    className="quote__mark"
  >
    ”
  </div>
  <blockquote
    cite="https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list"
    className="quote__content"
  >
    <p>
      Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.
    </p>
    <footer
      className="quote__author"
    >
      <cite>
        Stephen Covey
      </cite>
      <span>
        , 
      </span>
      <cite>
        My company
      </cite>
    </footer>
  </blockquote>
</div>
`;

exports[`<Quote /> Rendering should render correctly with no author 1`] = `
<div
  authorStyle={Object {}}
  className="quote-wrap quote"
  contentStyle={Object {}}
  quoteStyle={Object {}}
>
  <div
    className="quote__mark"
  >
    ”
  </div>
  <blockquote
    cite="https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list"
    className="quote__content"
  >
    <p>
      Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.
    </p>
    <footer
      className="quote__author"
    >
      <cite>
        My company
      </cite>
    </footer>
  </blockquote>
</div>
`;

exports[`<Quote /> Rendering should render correctly with no company 1`] = `
<div
  authorStyle={Object {}}
  className="quote-wrap quote"
  contentStyle={Object {}}
  quoteStyle={Object {}}
>
  <div
    className="quote__mark"
  >
    ”
  </div>
  <blockquote
    cite="https://blog.hubspot.com/sales/18-motivational-quotes-to-start-your-day-list"
    className="quote__content"
  >
    <p>
      Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment. Be patient with yourself. Self-growth is tender; it's holy ground. There's no greater investment.
    </p>
    <footer
      className="quote__author"
    >
      <cite>
        Stephen Covey
      </cite>
    </footer>
  </blockquote>
</div>
`;
